cnd/src/db/integration_tests.rs

```diff
@@ -1,56 +1,116 @@
-                fn prop(swap_id: Quickcheck<SwapId>,
-                    request: Quickcheck<Request<$alpha_ledger, $beta_ledger, $alpha_asset, $beta_asset>>,
-                    accept: Quickcheck<Accept<$alpha_ledger, $beta_ledger>>) -> anyhow::Result<bool> {
-                let db_path = tempfile::Builder::new()
+                fn prop(swap: Quickcheck<Swap>,
+                        request: Quickcheck<Request<$alpha_ledger, $beta_ledger, $alpha_asset, $beta_asset>>,
+                        accept: Quickcheck<Accept<$alpha_ledger, $beta_ledger>>,
+                ) -> anyhow::Result<bool> {
+
+                    // unpack the swap from the generic newtype
+                    let Swap { swap_id, role, counterparty } = swap.0;
+
+                    // construct the expected swap types from the function we get passed in order to enrich it with the role
+                    let expected_swap_types = ($expected_swap_types_fn)(role);
+
+                    let db_path = tempfile::Builder::new()
                         .prefix(&swap_id.to_string())
                         .suffix(".sqlite")
                         .tempfile()?
                         .into_temp_path();
 
-                let db = Sqlite::new(&db_path)?;
+                    let db = Sqlite::new(&db_path)?;
 
-                let saved_request = Request {
-                    swap_id: *swap_id,
-                    ..*request
-                };
-                let saved_accept = Accept {
-                    swap_id: *swap_id,
-                    ..*accept
-                };
+                    let saved_swap = Swap {
+                        swap_id,
+                        role,
+                        counterparty
+                    };
+                    let saved_request = Request {
+                        swap_id,
+                        ..*request
+                    };
+                    let saved_accept = Accept {
+                        swap_id,
+                        ..*accept
+                    };
 
-                db.save_message(saved_request.clone())?;
-                db.save_message(saved_accept.clone())?;
+                    db.save(saved_swap.clone())?;
+                    db.save_message(saved_request.clone())?;
+                    db.save_message(saved_accept.clone())?;
 
-                let (loaded_request, loaded_accept) = db.load_accepted_swap(*swap_id)?;
+                    let loaded_swap = Retrieve::get(&db, &swap_id)?;
+                    let (loaded_request, loaded_accept) = db.load_accepted_swap(swap_id)?;
+                    let loaded_swap_types = db.determine_types(&swap_id)?;
 
-                Ok(saved_request == loaded_request && saved_accept == loaded_accept)
-            }
+                    Ok(
+                        saved_request == loaded_request &&
+                        saved_accept == loaded_accept &&
+                        loaded_swap == saved_swap &&
```

Comment:
This could be a bit more homogenous (`saved` on the left and `loaded` on the right and not using `expected` terminology)

Reply:
I agree, I do this fix in a follow up PR, thanks.